src/test/java/io/openliberty/tools/intellij/it/GradleSingleModMPSIDProjectTest.java

```diff
@@ -0,0 +1,252 @@
+     */
+    private static final String SM_MP_PROJECT_NAME = "singleModGradleMP";
+
+    /**
+     * Single module Microprofile project name with space.
+     */
+    private static final String SM_MP_PROJECT_NAME_NEW = "singleMod GradleMP";
+
+    /**
+     * The path to the folder containing the test projects.
+     */
+    private static final String PROJECTS_PATH = Paths.get("src", "test", "resources", "projects", "gradle").toAbsolutePath().toString();
+
+    /**
+     * The path to the folder containing the test projects, including directories with spaces.
+     */
+    private static final String PROJECTS_PATH_NEW = Paths.get("src", "test", "resources", "projects", "gradle sample").toAbsolutePath().toString();
+
+    /**
+     * Project port.
+     */
+    private final int SM_MP_PROJECT_PORT = 9080;
+
+    /**
+     * Project resource URI.
+     */
+    private final String SM_MP_PROJECT_RES_URI = "api/resource";
+
+    /**
+     * Project response.
+     */
+    private final String SM_MP_PROJECT_OUTPUT = "Hello! Welcome to Open Liberty";
+
+    /**
+     * Relative location of the WLP installation.
+     */
+    private final String WLP_INSTALL_PATH = "build";
+
+    /**
+     * The path to the test report.
+     */
+    private final Path TEST_REPORT_PATH = Paths.get(PROJECTS_PATH_NEW, SM_MP_PROJECT_NAME_NEW, "build", "reports", "tests", "test", "index.html");
+
+    /**
+     * Build file name.
+     */
+    private final String BUILD_FILE_NAME = "build.gradle";
+
+    /**
+     * Action command to open the build file.
+     */
+    private final String BUILD_FILE_OPEN_CMD = "Liberty: View Gradle config";
+
+    /**
+     * Dev mode configuration start parameters.
+     */
+    private final String DEV_MODE_START_PARAMS = "--hotTests";
+
+    /**
+     * Dev mode configuration custom start parameters for debugging.
+     */
+    private final String DEV_MODE_START_PARAMS_DEBUG = "--libertyDebugPort=9876";
+
+    /**
+     * Prepares the environment for test execution.
+     */
+    @BeforeAll
+    public static void setup() throws IOException {
+        // Copy the directory from PROJECTS_PATH to PROJECTS_PATH_NEW
+        TestUtils.copyDirectory(PROJECTS_PATH, PROJECTS_PATH_NEW);
+
+        Path pathNew = Path.of(PROJECTS_PATH_NEW);
+        Path projectDirPath = pathNew.resolve(SM_MP_PROJECT_NAME);
+
+        // Define paths for the original and copy of settings.gradle
+        Path originalPath = projectDirPath.resolve("settings.gradle");
+        Path originalPathCopy = projectDirPath.resolve("settings-copy.gradle");
+
+        // Rename settings.gradle to settings-duplicate.gradle
+        Files.move(originalPath, originalPath.resolveSibling("settings-duplicate.gradle"));
+        // Rename settings-copy.gradle to settings.gradle
+        Files.move(originalPathCopy, originalPathCopy.resolveSibling("settings.gradle"));
+
+        Path projectDirNewPath = pathNew.resolve(SM_MP_PROJECT_NAME_NEW);
+
+        try {
```

Comment:
I don't understand the exception handling strategy here. Do you really just want to print a message when the directory change is made but not on other errors? It seems like the three lines above this should also be in this try statement and the error message adjusted. You probably should also use an assertion to terminate the test if there is an exception.

Reply:
I made the changes according to your suggestions and tested them. Everything was successful.